### ISSUE TITLE
Installing gersemi in python 3.9.11 from PyPI brings in the dataclasses package as a dependency which breaks both pip and gersemi

### ISSUE THREAD

**Issue by nlebedenco** [BlankSpruce/gersemi#36]
Installing gersemi via `pip install gersemi`  is bringing the dataclasses package (https://pypi.org/project/dataclasses/) on Windows hosts tested with both python 3.9 and 3.11. This does not seem to be a problem in python 3.11 but in python 3.9 it breaks both gersemi and pip and the only workaround is to manually remove the dataclasses package from the site-packages folder. That begs the question why this is happening since the dataclasses package seems to be targeted to python versions prior to 3.7 and should not be needed on python 3.7+ . 

This issue seems to be related to https://github.com/huggingface/transformers/issues/8638

```
C:\>pip install gersemi
Requirement already satisfied: gersemi in c:\ncs\toolchains\ce3b5ff664\opt\bin\lib\site-packages (0.15.1)
Requirement already satisfied: appdirs in c:\ncs\toolchains\ce3b5ff664\opt\bin\lib\site-packages (from gersemi) (1.4.4)
Collecting dataclasses (from gersemi)
  Using cached dataclasses-0.6-py3-none-any.whl.metadata (3.0 kB)
Requirement already satisfied: lark>=1.0 in c:\ncs\toolchains\ce3b5ff664\opt\bin\lib\site-packages (from gersemi) (1.1.7)
Requirement already satisfied: pyyaml<7,>=5 in c:\ncs\toolchains\ce3b5ff664\opt\bin\lib\site-packages (from gersemi) (6.0.1)
Using cached dataclasses-0.6-py3-none-any.whl (14 kB)
Installing collected packages: dataclasses
Successfully installed dataclasses-0.6

C:\>pip list
Traceback (most recent call last):
  File "runpy.py", line 197, in _run_module_as_main
  File "runpy.py", line 87, in _run_code
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Scripts\pip.exe\__main__.py", line 4, in <module>
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Lib\site-packages\pip\_internal\cli\main.py", line 11, in <module>
    from pip._internal.cli.autocompletion import autocomplete
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Lib\site-packages\pip\_internal\cli\autocompletion.py", line 10, in <module>
    from pip._internal.cli.main_parser import create_main_parser
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Lib\site-packages\pip\_internal\cli\main_parser.py", line 9, in <module>
    from pip._internal.build_env import get_runnable_pip
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Lib\site-packages\pip\_internal\build_env.py", line 18, in <module>
    from pip._internal.cli.spinners import open_spinner
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Lib\site-packages\pip\_internal\cli\spinners.py", line 9, in <module>
    from pip._internal.utils.logging import get_indentation
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Lib\site-packages\pip\_internal\utils\logging.py", line 13, in <module>
    from pip._vendor.rich.console import (
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Lib\site-packages\pip\_vendor\rich\console.py", line 60, in <module>
    from .pretty import Pretty, is_expandable
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Lib\site-packages\pip\_vendor\rich\pretty.py", line 390, in <module>
    class Node:
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Lib\site-packages\dataclasses.py", line 958, in dataclass
    return wrap(_cls)
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Lib\site-packages\dataclasses.py", line 950, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash, frozen)
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Lib\site-packages\dataclasses.py", line 800, in _process_class
    cls_fields = [_get_field(cls, name, type)
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Lib\site-packages\dataclasses.py", line 800, in <listcomp>
    cls_fields = [_get_field(cls, name, type)
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Lib\site-packages\dataclasses.py", line 659, in _get_field
    if (_is_classvar(a_type, typing)
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Lib\site-packages\dataclasses.py", line 550, in _is_classvar
    return type(a_type) is typing._ClassVar
AttributeError: module 'typing' has no attribute '_ClassVar'

C:\>gersemi --version 
Traceback (most recent call last):
  File "runpy.py", line 197, in _run_module_as_main
  File "runpy.py", line 87, in _run_code
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Scripts\gersemi.exe\__main__.py", line 4, in <module>
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Lib\site-packages\gersemi\__main__.py", line 6, in <module>
    from gersemi.configuration import (
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Lib\site-packages\gersemi\configuration.py", line 77, in <module>
    class Configuration:  # pylint: disable=too-many-instance-attributes
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Lib\site-packages\dataclasses.py", line 958, in dataclass
    return wrap(_cls)
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Lib\site-packages\dataclasses.py", line 950, in wrap
    return _process_class(cls, init, repr, eq, order, unsafe_hash, frozen)
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Lib\site-packages\dataclasses.py", line 800, in _process_class
    cls_fields = [_get_field(cls, name, type)
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Lib\site-packages\dataclasses.py", line 800, in <listcomp>
    cls_fields = [_get_field(cls, name, type)
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Lib\site-packages\dataclasses.py", line 659, in _get_field
    if (_is_classvar(a_type, typing)
  File "C:\ncs\toolchains\ce3b5ff664\opt\bin\Lib\site-packages\dataclasses.py", line 550, in _is_classvar
    return type(a_type) is typing._ClassVar
AttributeError: module 'typing' has no attribute '_ClassVar'
```

**Comment by BlankSpruce** [comment]
I'll release some fix later today. This might be good opportunity to trim supported Python versions.

**Comment by BlankSpruce** [comment]
I've released 0.16.0 but this will be rather short living version because I intent to tackle soon the other open issue and update formatter to upcoming CMake 3.31. Since 0.16.0 external `dataclasses` won't be necessary and the oldest supported Python version will be 3.8. This had reached end of life just a couple of days ago so I'm not axing that support yet. If there are any issues with 0.16.0 just let me know through new issue.
